crates/router/src/core/payments/transformers.rs

```diff
@@ -504,7 +505,33 @@ where
         ))
     }
 
-    let customer_details_response = customer.as_ref().map(ForeignInto::foreign_into);
+    // For the case when we don't have Customer data directly stored in Payment intent
+    let customer_table_response: Option<CustomerDetailsResponse> =
+        customer.as_ref().map(ForeignInto::foreign_into);
+
+    // If we have customer data in Payment Intent, We are populating the Retrieve response from the
+    // same
+    let customer_details_response =
+        if let Some(customer_details_raw) = payment_intent.customer_details.clone() {
+            let customer_details_encrypted =
+                serde_json::from_value::<CustomerData>(customer_details_raw.into_inner().expose());
+            let mut response_cus_data = CustomerDetailsResponse::default();
+            if let Ok(customer_details_encrypted_data) = customer_details_encrypted {
+                if let Some(customer_table_response) = customer_table_response {
+                    response_cus_data.set_id(customer_table_response.id);
+                }
+                response_cus_data.set_name(customer_details_encrypted_data.name);
+                response_cus_data.set_phone(customer_details_encrypted_data.phone);
+                response_cus_data.set_email(customer_details_encrypted_data.email);
+                response_cus_data
+                    .set_phone_country_code(customer_details_encrypted_data.phone_country_code);
+                Some(response_cus_data)
```

Comment:
we do not need a builder pattern here. Can use direct struct initialization. For more information where builder pattern will be useful you can refer to this article https://www.shuttle.rs/blog/2022/06/09/the-builder-pattern